sklearn/ensemble/tests/test_forest.py

```diff
@@ -1330,3 +1330,93 @@ def test_forest_degenerate_feature_importances():
     gbr = RandomForestRegressor(n_estimators=10).fit(X, y)
     assert_array_equal(gbr.feature_importances_,
                        np.zeros(10, dtype=np.float64))
+
+
+def test__get_n_bootstrap_samples():
```

Comment:
We usually don't test a private function. Instead we would test through the different estimator (RandomForest, ExtraTrees)

Reply:
You can parametrize easily this case and you can check. You can make a test function where want to check that errors are raised properly (you can parametrize as well).

The other behavior should be done by fitting the estimator and check that we fitted on a subset of data.

Reply:
> check that we fitted on a subset of data.

where do you get that info?

As far as I can see, the sample indices for each bootstrap are not stored anywhere, but translated into sample weights and passed to the estimator.

Reply:
(exception check unit tests refactored in: https://github.com/scikit-learn/scikit-learn/pull/14682/commits/081b7b7bc33d0a2b69493e34e4e60de0244c1c3b)

Reply:
> where do you get that info?

Yep I thought that we were having a private function to get back the indices but it does not look like so simple to do without making some hacks. I would need a bit more time to think about how to test this.